src/Authenticator/JwtAuthenticator.php

```diff
@@ -0,0 +1,138 @@
+    {
+        parent::__construct($identifiers, $config);
+
+        if (empty($this->_config['salt'])) {
+            if (class_exists('\Cake\Utility\Security')) {
+                $this->getConfig('salt', \Cake\Utility\Security::salt());
+            } else {
+                throw new RuntimeException('You must set the `salt` config key');
+            }
+        }
+    }
+
+    /**
+     * Authenticates the identity contained in a request. Will use the `config.userModel`, and `config.fields`
+     * to find POST data that is used to find a matching record in the `config.userModel`. Will return false if
+     * there is no post data, either username or password is missing, or if the scope conditions have not been met.
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request The request that contains login information.
+     * @param \Psr\Http\Message\ResponseInterface $response Unused response object.
+     * @return \Authentication\ResultInterface
+     */
+    public function authenticate(ServerRequestInterface $request, ResponseInterface $response)
+    {
+        $result = $this->getPayload($request);
+
+        if (!$result instanceof stdClass) {
+            return new Result(null, Result::FAILURE_CREDENTIAL_INVALID);
+        }
+
+        $result = json_decode(json_encode($result), true);
+
+        if (empty($result)) {
+            return new Result(null, Result::FAILURE_CREDENTIALS_NOT_FOUND);
+        }
+
+        if ($this->getConfig('returnPayload')) {
+            $entityClass = $this->getConfig('entityClass');
+            $entity = new $entityClass($result);
+
+            return new Result($entity, Result::SUCCESS);
+        }
+
+        $user = $this->identifiers()->identify($result);
+
+        if (empty($user)) {
+            return new Result(null, Result::FAILURE_IDENTITY_NOT_FOUND, $this->identifiers()->getErrors());
+        }
+
+        return new Result($user, Result::SUCCESS);
+    }
+
+    /**
+     * Get payload data.
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface|null $request Request to get authentication information from.
+     * @return object|null Payload object on success, null on failure
+     */
+    public function getPayload(ServerRequestInterface $request = null)
+    {
+        if (!$request) {
+            return $this->_payload;
+        }
+
+        $payload = null;
+        $token = $this->_getToken($request);
+
+        if ($token) {
+            $payload = $this->_decodeToken($token);
+        }
+
+        return $this->_payload = $payload;
+    }
+
+    /**
+     * Decode JWT token.
+     *
+     * @param string $token JWT token to decode.
+     * @return object|null The JWT's payload as a PHP object, null on failure.
+     */
+    protected function _decodeToken($token)
+    {
+        $config = $this->getConfig();
+        $token = str_ireplace($config['tokenPrefix'] . ' ', '', $token);
```

Comment:
I haven't ever seen a token with words in it, except there is a language that is made of random characters :)